apps/client/src/features/app-settings/panel/project-panel/ProjectListItem.tsx

```diff
@@ -0,0 +1,176 @@
+import { useCallback, useState } from 'react';
+import { IconButton, Menu, MenuButton, MenuItem, MenuList } from '@chakra-ui/react';
+import { IoEllipsisHorizontal } from '@react-icons/all-files/io5/IoEllipsisHorizontal';
+import { AxiosError } from 'axios';
+
+import { PROJECT_LIST } from '../../../../common/api/apiConstants';
+import { invalidateAllCaches } from '../../../../common/api/apiUtils';
+import { duplicateProject, loadProject, renameProject } from '../../../../common/api/ontimeApi';
+import { ontimeQueryClient } from '../../../../common/queryClient';
+
+import DuplicateRenameProjectForm, { DuplicateRenameProjectFormValues } from './DuplicateRenameProjectForm';
+import { EditMode } from './ProjectList';
+
+import style from './ProjectPanel.module.scss';
+
+interface ProjectListItemProps {
+  current?: boolean;
+  filename: string;
+  createdAt: string;
+  updatedAt: string;
+  onToggleEditMode: (editMode: EditMode, filename: string | null) => void;
+  onSubmit: () => void;
+  editingFilename: string | null;
+  editingMode: EditMode | null;
+}
+
+export default function ProjectListItem({
+  current,
+  createdAt,
+  editingFilename,
+  editingMode,
+  filename,
+  onSubmit,
+  onToggleEditMode,
+  updatedAt,
+}: ProjectListItemProps) {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+
+  const handleRefetch = async () => {
+    await ontimeQueryClient.invalidateQueries({ queryKey: PROJECT_LIST });
+  };
+
+  const handleSubmitRename = useCallback(
+    async (values: DuplicateRenameProjectFormValues) => {
+      try {
+        setSubmitError(null);
+
+        if (!values.filename) {
+          setSubmitError('Filename cannot be blank');
+          return;
+        }
+        await renameProject(filename, values.filename);
+        await handleRefetch();
+        onSubmit();
+      } catch (error) {
+        if (error instanceof AxiosError) {
+          const errorMessage = error?.response?.data?.message;
+
+          setSubmitError(errorMessage);
+        } else {
+          setSubmitError('An unknown error occurred');
+        }
+      }
+    },
+    [filename, onSubmit],
+  );
+
+  const handleSubmitDuplicate = useCallback(
+    async (values: DuplicateRenameProjectFormValues) => {
+      try {
+        setSubmitError(null);
+
+        if (!values.filename) {
+          setSubmitError('Filename cannot be blank');
+          return;
+        }
+        await duplicateProject(filename, values.filename);
+        await handleRefetch();
+        onSubmit();
+      } catch (error) {
+        if (error instanceof AxiosError) {
+          const errorMessage = error?.response?.data?.message;
+
+          setSubmitError(errorMessage);
+        } else {
+          setSubmitError('An unknown error occurred');
+        }
+      }
+    },
+    [filename, onSubmit],
+  );
+
+  const handleToggleEditMode = useCallback(
```

Comment:
question: what is the value of `useCallback` here?

Since the function is consumed locally we might not be doing much more than paying the cost of the memoisation check?

Reply:
`The 'handleToggleEditMode' function makes the dependencies of useCallback Hook (at line 102) change on every render. To fix this, wrap the definition of 'handleToggleEditMode' in its own useCallback() Hook.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)`

that line is another `useCallback`, but if I remove that one then we get the warning for the `useMemo`